### PR TITLE
Photospline stylefixes

### PR DESCRIPTION
**Pull request by HansN87**
follow up to https://github.com/IceCubeOpenSource/skyllh/issues/3 and https://github.com/IceCubeOpenSource/skyllh/pull/8

### CODE REVIEW

File: skyllh/core/utils/multidimgridpdf.py [hunk 1]
```diff
@@ -76,24 +77,24 @@ def create_MultiDimGridPDF_from_photosplinetable(
...
             kde_pdf_axis_name_map_inv[var], num_dict['bins'][idx])
-        for (idx,var) in enumerate(num_dict['vars'])
+        for (idx, var) in enumerate(num_dict['vars'])
     ]
- 
-    # get splinetable file
-    splinetable_file_list = ds.get_aux_data_definition(splinetable_key) 
-    # this is a list with only one element
+
+    # Getthe splinetable file.
```
Comment (HansN87) [inline skyllh/core/utils/multidimgridpdf.py]: whitespace after Get

File: skyllh/core/pdf.py [hunk 2]
```diff
@@ -675,26 +691,31 @@ def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=No
                 vmax=axis_binning.upper_edge
             ))
 
-       
-        pdf_loaded = False
+        # need either splinetable or value grid
+        if((pdf_path_to_splinetable is None) and (pdf_grid_data is None)):
```
Comment (HansN87) [inline skyllh/core/pdf.py]: better: path_to_pdf_splinetable

File: skyllh/core/pdf.py [hunk 3]
```diff
@@ -675,26 +691,31 @@ def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=No
...
-                self._pdf = photospline.SplineTable(pdf_path_to_splinetable)
-                pdf_loaded = True
-                self._photospline = True
+        else:
+            if(isinstance(pdf_path_to_splinetable, str) and not PHOTOSPLINE_LOADED):
+                raise ImportError(
+                    "missing library: can not find photosplines.")
 
+            if not isinstance(pdf_grid_data, np.ndarray):
+                raise TypeError("missing or wrong input: can not create RegularGridInterpolator from type {}".format(
```
Comment (HansN87) [inline skyllh/core/pdf.py]: add expected type to error message

File: skyllh/core/backgroundpdf.py [hunk 4]
```diff
@@ -16,7 +16,8 @@ class BackgroundMultiDimGridPDF(MultiDimGridPDF, IsBackgroundPDF):
     interpolated using a :class:`scipy.interpolate.RegularGridInterpolator`
     instance.
     """
-    def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=None, norm_factor_func=None): 
+
+    def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=None, norm_factor_func=None):
```
Comment (chrhck) [inline skyllh/core/backgroundpdf.py]:
```suggestion
    def __init__(
        self,
        axis_binnings,
        pdf_path_to_splinetable=None,
        pdf_grid_data=None,
        norm_factor_func=None):
```

File: skyllh/core/backgroundpdf.py [hunk 5]
```diff
@@ -29,8 +30,9 @@ def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=No
             the PDF axes. The name of each BinningDefinition instance defines
             the event field name that should be used for querying the PDF.
         pdf_path_to_splinetable : str
-            The path to the file that contains the spline table
-            (a pre-computed fit to pdf_grid_data)
+            The path to the file containing the spline table.
+            The spline table contains a  pre-computed fit to pdf_grid_data.
```
Comment (chrhck) [inline skyllh/core/backgroundpdf.py]:
```suggestion
            The spline table contains a pre-computed fit to pdf_grid_data.
```

File: skyllh/core/pdf.py [hunk 6]
```diff
@@ -646,8 +663,8 @@ def __init__(self, axis_binnings, pdf_path_to_splinetable=None, pdf_grid_data=No
             the PDF axes. The name of each BinningDefinition instance defines
             the event field name that should be used for querying the PDF.
         pdf_path_to_splinetable : str
-            The path to the file that contains the spline table
-            (a pre-computed fit to pdf_grid_data)
+            The path to the file containing the spline table.
+            The spline table contains a  pre-computed fit to pdf_grid_data.
```
Comment (chrhck) [inline skyllh/core/pdf.py]:
```suggestion
            The spline table contains a pre-computed fit to pdf_grid_data.
```

File: skyllh/core/utils/multidimgridpdf.py [hunk 7]
```diff
@@ -31,11 +32,12 @@ def kde_pdf_bkg_norm_factor_func(pdf, tdm, fitparams):
...
     return 1. / (2 * np.pi)
 
+
 def create_MultiDimGridPDF_from_photosplinetable(
         ds, data, info_key, splinetable_key, norm_factor_func=None,
         kind=None, tl=None):
-    """Creates a MultiDimGridPDF instance with pdf values taken from KDE PDF
-    values stored in the dataset's auxiliary data.
+    """Creates a MultiDimGridPDF instance with pdf values taken from photospline pdf,
+    a spline interpolation of KDE PDF values stored in a splinetable on disk.
```
Comment (chrhck) [inline skyllh/core/utils/multidimgridpdf.py]:
```suggestion
    """
    Creates a MultiDimGridPDF instance with pdf values taken from photospline pdf,
    a spline interpolation of KDE PDF values stored in a splinetable on disk.
```

File: skyllh/core/utils/multidimgridpdf.py (hunk 1 above)
Comment (chrhck) [inline skyllh/core/utils/multidimgridpdf.py]:
```suggestion
    # Get the splinetable file.
```

File: skyllh/core/pdf.py (hunk 2 above)
Comment (HansN87) [reply skyllh/core/pdf.py]: resolved in https://github.com/IceCubeOpenSource/skyllh/pull/22/commits/d33951c1cff6af86a1771b651c729295daee7fde

File: skyllh/core/pdf.py (hunk 3 above)
Comment (HansN87) [reply skyllh/core/pdf.py]: resolved in https://github.com/IceCubeOpenSource/skyllh/pull/22/commits/d33951c1cff6af86a1771b651c729295daee7fde

**Comment by chrhck** [comment]
It might be better to just compare changes & merge offline, since the github diff seems to be a bit messed up. locally the diff between this branch and master shows only 4 changes files (as it should)

**Comment by HansN87** [comment]
> It might be better to just compare changes & merge offline, since the github diff seems to be a bit messed up. locally the diff between this branch and master shows only 4 changes files (as it should)

indeed. git diff master..photospline_stylefixes behaves as expected (i.e. does not show the rebased changes from unrelated commits to master.)

**Pull request merged**